test/functional/ui/045_folding_spec.lua

```diff
@@ -0,0 +1,218 @@
+
+    screen:expect([[
+        dd {{{            |
+        ee {{{ }}}        |
+      {{{                 |
+        ff }}}            |
+        ff }}}            |
+      ^                    |
+      line 2 foldlevel=2  |
+      1                   |
+      1                   |
+                          |
+    ]])
+
+  end)
+
+  it("Test folding with indent.", function()
+    screen:try_resize(20, 8)
+    execute('set fdm=indent sw=2')
+    insert([[
+    aa
+      bb
+        cc
+    last
+    ]])
+    execute('call append("$", "foldlevel line3=" . foldlevel(3))')
+    execute('call append("$", foldlevel(2))')
+    feed('zR')
+
+    screen:expect([[
+      aa                  |
+        bb                |
+          cc              |
+      last                |
+      ^                    |
+      foldlevel line3=2   |
+      1                   |
+                          |
+    ]])
+  end)
+
+  it("Test syntax folding", function()
+    screen:try_resize(35, 15)
+    insert([[
+      1 aa
+      2 bb
+      3 cc
+      4 dd {{{
+      5 ee {{{ }}}
+      6 ff }}}
+      7 gg
+      8 hh
+      9 ii
+      a jj
+      b kk
+      last]])
+    execute('set fdm=syntax fdl=0')
+    execute('syn region Hup start="dd" end="ii" fold contains=Fd1,Fd2,Fd3')
+    execute('syn region Fd1 start="ee" end="ff" fold contained')
+    execute('syn region Fd2 start="gg" end="hh" fold contained')
+    execute('syn region Fd3 start="commentstart" end="commentend" fold contained')
+    feed('Gzk')
+    execute('call append("$", "folding " . getline("."))')
+    feed('k')
+    execute('call append("$", getline("."))')
+    feed('jAcommentstart  <esc>Acommentend<esc>')
+    execute('set fdl=1')
+    feed('3j')
+    execute('call append("$", getline("."))')
+    execute('set fdl=0')
+    feed('zO<C-L>j') -- <C-L> redraws screen
+    execute('call append("$", getline("."))')
+    execute('set fdl=0')
+    expect([[
+      folding 9 ii
+      3 cc
+      9 ii
+      a jj]], true)
+  end)
+
+  it("Test expression folding.", function()
+    insert([[
+      1 aa
+      2 bb
+      3 cc
+      4 dd {{{
+      5 ee {{{ }}}
+      6 ff }}}
+      7 gg
+      8 hh
+      9 ii
+      a jj
+      b kk
+      last ]])
+
+    execute([[
+    fun Flvl()
+     let l = getline(v:lnum)
+     if l =~ "bb$"
+       return 2
+     elseif l =~ "gg$"
+       return "s1"
+     elseif l =~ "ii$"
+       return ">2"
+     elseif l =~ "kk$"
+       return "0"
+     endif
+     return "="
+    endfun
+    ]])
+    execute('set fdm=expr fde=Flvl()')
+    execute('/bb$')
+    execute('call append("$", "expr " . foldlevel("."))')
+    execute('/hh$')
+    execute('call append("$", foldlevel("."))')
+    execute('/ii$')
+    execute('call append("$", foldlevel("."))')
+    execute('/kk$')
+    execute('call append("$", foldlevel("."))')
+
+    expect([[
+      expr 2
+      1
+      2
+      0]], true)
+  end)
+
+  it('can open after fold after :move', function()
```

Comment:
immediately below this line, add this comment:

    -- luacheck: ignore

and that will disable luacheck for this `it(...)` scope.